docs/source/kv_cache/mooncake.rst

```diff
@@ -1,11 +1,226 @@
+         -d '{
+           "model": "meta-llama/Llama-3.1-70B-Instruct",
+           "prompt": "The future of AI is",
+           "max_tokens": 100,
+           "temperature": 0.7
+         }'
+
+**Debugging Tips:**
+
+1. **Enable verbose logging:**
+
+   .. code-block:: bash
+
+       mooncake_master -v=1
+
+2. **Check service status:**
+
+   .. code-block:: bash
+
+       # Check if services are running
+       ps aux | grep mooncake
+       netstat -tlnp | grep -E "(8080|50051)"
+
+3. **Monitor metrics:**
+
+   Access metrics at ``http://localhost:9003`` when master service is running.
+
+Configuration
+-------------
+
+**LMCache Parameters:**
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 15 60
+
+   * - Parameter
+     - Default
+     - Description
+   * - ``chunk_size``
+     - 256
+     - Number of tokens per KV chunk
+   * - ``local_device``
+     - "cpu"
+     - Local storage device type
+   * - ``remote_url``
+     - Required
+     - MoonCake store connection URL (format: ``mooncakestore://host:port/``)
+   * - ``remote_serde``
+     - "naive"
+     - Serialization method for remote storage
+   * - ``local_cpu``
+     - False
+     - Enable/disable local CPU caching (set to False for pure MoonCake evaluation)
+   * - ``max_local_cpu_size``
+     - Required
+     - Maximum local CPU cache size in GB (required even when local_cpu is False)
+
+**MoonCake Parameters (via extra_config):**
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 15 60
+
+   * - Parameter
+     - Default
+     - Description
+   * - ``local_hostname``
+     - Required
+     - Hostname/IP of the local node for MoonCake client identification
+   * - ``metadata_server``
+     - Required
+     - Address of metadata coordination server (etcd/Redis/HTTP format)
+   * - ``master_server_address``
+     - Required
+     - MoonCake master service address (host:port format)
+   * - ``protocol``
+     - "tcp"
+     - Communication protocol ("rdma" for high performance, "tcp" for compatibility)
+   * - ``device_name``
+     - ""
+     - RDMA device specification (e.g., "erdma_0,erdma_1" or "mlx5_0,mlx5_1")
+   * - ``global_segment_size``
+     - 3355443200
+     - **Memory size contributed by each vLLM worker** in bytes (~3.1GB)
+   * - ``local_buffer_size``
+     - 1073741824
+     - Local buffer allocation size in bytes (~1GB)
+   * - ``transfer_timeout``
+     - 1
+     - Timeout for transfer operations in seconds
+
+.. important::
+   **Understanding global_segment_size**: This parameter defines the amount of memory each vLLM worker contributes to the distributed memory pool. 
+   The total cluster memory available for KV cache storage will be: ``number_of_vllm_workers × global_segment_size``.
+   
+   Adjust this value based on your available system memory and expected cache requirements.
+
+Additional Resources
+--------------------
+
+- `MoonCake Store Architecture <https://github.com/kvcache-ai/Mooncake/blob/main/doc/en/mooncake-store-preview.md>`_
```

Comment:
As Mooncake is a word already, so it could be better to use `Mooncake` here

Reply:
Thks. Resolved